[rmj1405] iP

rmj1405 opened the pull request:
### RubiRyo ChatBot 🐶

- Do you struggle with managing your tasks?
- Have _no motivation_ to be working by yourself?
- Feel **lonely**

RubiRyo ChatBox is what you need to use!

1. Inbuilt task management with various types such as Todo, Event, Deadlines
2. Friendly interface and tone, it doesn't feel like you're talking to an app

- [ ] I will download it!
- [ ] I love this app!
- [ ] RubiRyo 🥰

DOWNLOAD [HERE
](https://images.app.goo.gl/WvQY4Gw6bpMMPD1x5)

What our customers say:
> RubiRyo has changed my life, try it and turn your life around!


If you  are Java programmer, you can use it to practice Java too. Here's the `main` method:

```Java
public static void main(String[] args) {
    WelcomeUI welcome = new WelcomeUI();
    Scanner input = new Scanner(System.in);
    ChatBot chat = new ChatBot(input, welcome);

    chat.run();
    input.close();
}
```

Commit:
Add Gradle support

rmj1405 committed:
Customised Duke and greet,echo,exit functionality

rmj1405 committed:
Added insert to list and display functionality

rmj1405 committed:
Added new classes to organise program and fulfill check task feature

rmj1405 committed:
Added todos, events and deadlines classes

rmj1405 committed:
Added delete task functionality

rmj1405 committed:
Added exception handling for user input errors

rmj1405 committed:
Updated input.txt and attempted regression testing

rmj1405 committed:
Temporary removed logo to facilitate testing

rmj1405 committed:
Restructured project into packages

rmj1405 committed:
Update chatbot to take in user inputs with "/" for deadline and event tasks

rmj1405 committed:
Add more packages for improved structure

rmj1405 committed:
Refactor task file names

rmj1405 committed:
Add read date and time from user input feature

rmj1405 committed:
Attempt to implement save to hardisk feature

rmj1405 committed:
Update task manager to write to hardisk

rmj1405 committed:
Abstract commands into separate classes and add common parser methods

rmj1405 committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

rmj1405 committed:
Set up gradle

rmj1405 committed:
Add skeleton for JUnit tests

rmj1405 committed:
Add JUnit test for DeadlineCommand class

rmj1405 committed:
Merge branch 'branch-Level-8'

rmj1405 committed:
Standardise exceptions for invalid date and time and missing user inputs to DukeException

rmj1405 committed:
Add save to hardisk feature when app is shutdown

rmj1405 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/ChatBot.java
#	src/main/java/task/Deadline.java
#	src/main/java/task/Event.java
#	src/main/java/task/TaskManager.java

Resolved

rmj1405 committed:
Bug fixes for Level-7 after merging with master branch

rmj1405 committed:
Add JUnit tests for DeadlineCommand and EventCommand classes

rmj1405 committed:
.gitignore: Add MANIFEST.MF

MANIFEST.MF created when app is packaged as .jar file

rmj1405 committed:
Add JavaDocs for all all non-private and non-trivial classes/methods

rmj1405 committed:
Tweak code to adhere to the coding standards

rmj1405 committed:
Add find task by searching for keyword feature

rmj1405 committed:
Merge branch 'branch-A-JavaDoc'

rmj1405 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/command/UncheckCommand.java
#	src/main/java/task/ToDo.java

Resolved

rmj1405 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/ChatBot.java

Resolved.

rmj1405 committed:
Add and configure checkstyle

rmj1405 committed:
Merge branch 'branch-A-CheckStyle'

rmj1405 committed:
Add preliminary JavaFX

rmj1405 committed:
Restructure code to follow JavaFX tutorial

rmj1405 committed:
Bug fixes after code restructure

leeyi45 commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,51 @@
+    public static boolean isNumeric(String string) {
+        int intValue;
+
+        if (string == null || string.equals("")) {
```
Check out String.isEmpty

leeyi45 commented on src/main/java/task/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+     * Marks the task as completed.
+     * @param index
+     */
+    public void checkTask(int index) {
```
Check is potentially ambiguous: check the state of a task, or mark the task as done

wengkit1 replied:
Could check be different verb?

rmj1405 replied:
thanks for pointing that out!

leeyi45 commented on src/main/java/command/CheckCommand.java:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Executes check task in list command.
+ */
+public class CheckCommand extends Command {
```
Check is ambiguous....

rmj1405 committed:
Save basic chatbot GUI - JavaFX tutorial 3

clydelhui commented on src/main/java/ChatBot.java:
```diff
@@ -0,0 +1,109 @@
+                    continue;
+                case "list":
+                    ListCommand lc = new ListCommand(taskManager);
+                    lc.executeCommand();
+                    break;
+                case "bye":
+                    ByeCommand bc = new ByeCommand();
+                    bc.executeCommand();
+                    fileManager.saveTasksToFile(taskManager);
+                    break loop;
+                case "find":
+                    FindCommand fc = new FindCommand(taskManager, inputArr[1]);
+                    fc.executeCommand();
+                    break;
+                case "check":
+                    CheckCommand cc = new CheckCommand(taskManager, inputArr[1]);
+                    cc.executeCommand();
+                    break;
+                case "uncheck":
+                    UncheckCommand uc = new UncheckCommand(taskManager, inputArr[1]);
+                    uc.executeCommand();
+                    break;
+                case "delete":
+                    DeleteCommand dc = new DeleteCommand(taskManager, inputArr[1]);
+                    dc.executeCommand();
+                    break;
+                case "todo":
+                    TodoCommand tc = new TodoCommand(taskManager, inputArr[1]);
+                    tc.executeCommand();
+                    break;
+                case "event":
+                    EventCommand ec = new EventCommand(taskManager, inputArr[1]);
+                    ec.executeCommand();
+                    break;
+                case "deadline":
+                    DeadlineCommand dlc = new DeadlineCommand(taskManager, inputArr[1]);
+                    dlc.executeCommand();
+                default:
+                    throw new DukeException("Command not recognised, please enter a valid command!");
+                }
```
Good that you used the indent for `switch case` according to the coding standard! 👍

wengkit1 replied:
Nice to include default behaviour in switch.

clydelhui commented on src/main/java/ChatBot.java:
```diff
@@ -0,0 +1,109 @@
+                    lc.executeCommand();
+                    break;
+                case "bye":
+                    ByeCommand bc = new ByeCommand();
+                    bc.executeCommand();
+                    fileManager.saveTasksToFile(taskManager);
+                    break loop;
```
Since you override the `executeCommand()` method in all child classes of `Command`, perhaps you can utilise polymorphism and call the method outside the `switch case` block?

Perhaps you could also consider exiting the program within the `byeCommand` 's implementation of `executeCommand()`?

clydelhui commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * Abstract class that executes the command given by the user.
+ */
+public abstract class Command {
```
I like that you made this an abstract class to ensure that a `Command` cannot be instantiated! 💯

bryanongjx replied:
agree with clyde, very good use of abstract classes!

clydelhui commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,98 @@
+ * the list. A task object has a completion status and
+ * a description.
+ */
+public class Task {
```
Perhaps you could consider making `Task` abstract as well?

clydelhui commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,98 @@
+    private String description;
+    private boolean isDone;
+
+    public Task(String description, boolean status) {
```
Is there a reason why the parameter is called `status` instead of `isDone`?

clydelhui reviewed (commented):
Neat code with appropriate comments that make the code very easy to follow! Method names are easy to understand! I like the design decision to make some classes abstract and the use of polymorphism!

wengkit1 commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+package task;
```
Could this be packaged with the bot's main functions?

rmj1405 replied:
Hi! Could I clarify what you mean by packaging with the bot's main functions? Thank you!

bryanongjx commented on src/main/java/util/FileManager.java:
```diff
@@ -0,0 +1,96 @@
+package util;
+
+import task.*;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.Serializable;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+import java.util.Scanner;
+import java.util.ArrayList;
```
good organisation of packages! easy to understand what you are using in this class!

bryanongjx commented on src/main/java/ChatBot.java:
```diff
@@ -0,0 +1,109 @@
+                    bc.executeCommand();
+                    fileManager.saveTasksToFile(taskManager);
+                    break loop;
+                case "find":
+                    FindCommand fc = new FindCommand(taskManager, inputArr[1]);
+                    fc.executeCommand();
+                    break;
+                case "check":
+                    CheckCommand cc = new CheckCommand(taskManager, inputArr[1]);
+                    cc.executeCommand();
+                    break;
+                case "uncheck":
+                    UncheckCommand uc = new UncheckCommand(taskManager, inputArr[1]);
+                    uc.executeCommand();
+                    break;
+                case "delete":
+                    DeleteCommand dc = new DeleteCommand(taskManager, inputArr[1]);
+                    dc.executeCommand();
+                    break;
+                case "todo":
+                    TodoCommand tc = new TodoCommand(taskManager, inputArr[1]);
+                    tc.executeCommand();
+                    break;
+                case "event":
+                    EventCommand ec = new EventCommand(taskManager, inputArr[1]);
+                    ec.executeCommand();
+                    break;
+                case "deadline":
+                    DeadlineCommand dlc = new DeadlineCommand(taskManager, inputArr[1]);
+                    dlc.executeCommand();
+                default:
+                    throw new DukeException("Command not recognised, please enter a valid command!");
+                }
```
clean use of cases!

bryanongjx commented on src/main/java/command/EventCommand.java:
```diff
@@ -0,0 +1,40 @@
+        this.description = description;
+    }
+
+    /**
```
comments are well organised and straight to the point! :)

bryanongjx reviewed (commented):
Overall, code seems very clean, with good use of abstractions and well written comments. Easily understandable

rmj1405 committed:
Add completed JavaFX GUI

rmj1405 committed:
Merge branch 'branch-Level-10'

rmj1405 committed:
Update main class name

rmj1405 committed:
Add assertions to make program more safe

Assertions are used to define assumptions about the program state so that the runtime can verify them. An assertion failure indicates a possible bug in the code because the code has resulted in a program state that violates an assumption about how the code should behave.

Using assertions ensures that possible bugs are easily detected and terminates execution with an error message.

rmj1405 committed:
Clean up and refactor code to adhere to code quality standards

Add and update javadocs to all necessary methods and classes.

Abstract out String return messages and add them as methods in DukeUI.class.

"Check" class and methods refactored to "Mark" class and methods to prevent ambiguity. "Uncheck" class and methods also refactored to "Unmark" class to prevent ambiguity.

Delete unused run and main methods in Duke.class to preserve clean code.

rmj1405 committed:
Set up CI with GitHub Actions

GitHub Actions makes it easy to automate all your software workflows, now with world-class CI/CD. Build, test, and deploy your code right from GitHub. Make code reviews, branch management, and issue triaging work the way you want.

rmj1405 committed:
Merge branch 'branch-A-Assertions'

rmj1405 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/command/CheckCommand.java
#	src/main/java/command/ListCommand.java
#	src/main/java/command/UncheckCommand.java
#	src/main/java/task/Deadline.java
#	src/main/java/task/Event.java
#	src/main/java/task/ToDo.java

Resolved

rmj1405 committed:
Delete old check and uncheck command files added due to merge conflict

rmj1405 committed:
Merge branch 'branch-A-CI'

rmj1405 committed:
Fix bugs in displaying tasks and task loading after merge with assertions

rmj1405 committed:
Add sort tasks features - WIP

rmj1405 committed:
Implement help menu extension

Changes extension from sorting to help.

rmj1405 committed:
Merge branch 'branch-BCD-Extension'

rmj1405 committed:
Improve GUI and update images

rmj1405 committed:
Remove redundant css

rmj1405 committed:
Update credit

rmj1405 committed:
Add UI screenshot

Also fixes no new line checkstyle error

rmj1405 committed:
Minor tweak to UI screenshot

rmj1405 committed:
Complete User Guide

rmj1405 committed:
Fix checkstyle errors in test files

rmj1405 committed:
Fix IOException in jar file when user tried to save tasks

rmj1405 committed:
Update archive base name
